Looks like the [same problem as always](https://forum.palemoon.org/viewtopic.php?f=5&t=28480). :confused:

```
78 0:34.37(B Traceback (most recent call last):
78 0:34.37(B   File "/build/source/platform/config/pythonpath.py", line 56, in <module>
78 0:34.37(B     main(sys.argv[1:])
78 0:34.37(B   File "/build/source/platform/config/pythonpath.py", line 48, in main
78 0:34.37(B     execfile(script, frozenglobals)
78 0:34.37(B   File "/build/source/platform/python/mozbuild/mozbuild/action/xpidl-process.py", line 20, in <module>
78 0:34.37(B     from xpidl.typelib import write_typelib
78 0:34.37(B EOFError: EOF read where object expected
78 0:34.37(B make[6]: *** [Makefile:45: ../../../dist/bin/browser/components/shellservice.xpt] Error 1
```

> Try powering through a little less excessively […] so python configure has a chance to write out the temporary configuration files before they are being used.

> Another potential issue is the related I/O when doing many parallel tasks, especially early in the build process where many files are being read and written by the build system. if you want to run many parallel tasks you must have very fast I/O as well.

--- a/pkgs/applications/networking/browsers/palemoon/default.nix
+++ b/pkgs/applications/networking/browsers/palemoon/default.nix
@@ -115,7 +115,7 @@ stdenv.mkDerivation rec {
 
     # Too many cores can lead to build flakiness
     # https://forum.palemoon.org/viewtopic.php?f=5&t=28480
-    export jobs=$(($NIX_BUILD_CORES<=20 ? $NIX_BUILD_CORES : 20))
+    export jobs=$(($NIX_BUILD_CORES<=16 ? $NIX_BUILD_CORES : 16))
     if [ -z "$enableParallelBuilding" ]; then
       jobs=1
     fi

